plugin/evm/vm_test.go

```diff
@@ -810,6 +817,192 @@ func TestIssueAtomicTxs(t *testing.T) {
+	}
+
+	// the atomic tx should be safe in blk2 for now
+	// and getAtomicTx should report as processing even though it got discarded
+	exportTxFetched, status, height, err := vm.getAtomicTx(exportTx.ID())
+	assert.NoError(t, err)
+	assert.Equal(t, Processing, status)
+	assert.Equal(t, uint64(2), height, "expected height of export tx to be 2")
+	assert.Equal(t, exportTxFetched.ID(), exportTx.ID(), "expected ID of fetched export tx to match original txID")
+
+	if err := blk2.Accept(context.Background()); err != nil {
+		t.Fatal(err)
+	}
+
+	if status := blk2.Status(); status != choices.Accepted {
+		t.Fatalf("Expected status of accepted block to be %s, but found %s", choices.Accepted, status)
+	}
+
+	if lastAcceptedID, err := vm.LastAccepted(context.Background()); err != nil {
+		t.Fatal(err)
+	} else if lastAcceptedID != blk2.ID() {
+		t.Fatalf("Expected last accepted blockID to be the accepted block: %s, but found %s", blk2.ID(), lastAcceptedID)
+	}
+
+	// The same atomic transaction is included in multiple processing blocks that were built at around the same time.
+	// It gets rejected in one block and is still processing in the other.
+	// getAtomicTx should return status Processing
+	exportTx2, err := vm.newExportTx(vm.ctx.AVAXAssetID, 1, vm.ctx.XChainID, testShortIDAddrs[0], initialBaseFee, []*secp256k1.PrivateKey{testKeys[0]})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = vm.mempool.AddLocalTx(exportTx2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	<-issuer
+
+	blk3, err := vm.BuildBlock(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// blk4 should have a slightly different timestamp
+	vm.clock.Set(vm.clock.Time().Add(1 * time.Second))
+
+	err = vm.mempool.ForceAddTx(exportTx2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	<-issuer
+
+	blk4, err := vm.BuildBlock(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = blk3.Verify(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = blk4.Verify(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if status := blk3.Status(); status != choices.Processing {
+		t.Fatalf("Expected status of accepted block to be %s, but found %s", choices.Processing, status)
+	}
+	if status := blk4.Status(); status != choices.Processing {
+		t.Fatalf("Expected status of accepted block to be %s, but found %s", choices.Processing, status)
+	}
+
+	if err := blk3.Reject(context.Background()); err != nil {
+		t.Fatal(err)
+	}
+
+	// getAtomicTx should still return status Processing, even though blk3 got rejected with exportTx2, since blk4 is still processing with exportTx2.
+	exportTxFetched, status, height, err = vm.getAtomicTx(exportTx2.ID())
+	assert.NoError(t, err)
+	assert.Equal(t, Processing, status)
+	assert.Equal(t, uint64(3), height, "expected height of export tx to be 3")
+	assert.Equal(t, exportTxFetched.ID(), exportTx2.ID(), "expected ID of fetched export tx to match original txID")
```

Comment:
Can we add a final check here to accept `blk4` and check the status of `exportTx2` is correctly indexed and marked as accepted?

Reply:
done